shared/src/commonMain/kotlin/co/nimblehq/kmm/template/data/extensions/ResponseMapping.kt

```diff
@@ -0,0 +1,11 @@
+package co.nimblehq.kmm.template.data.extensions
+
+import kotlinx.coroutines.flow.*
+import kotlin.experimental.ExperimentalTypeInference
+
+@OptIn(ExperimentalTypeInference::class)
+fun <T> flowTransform(@BuilderInference block: suspend FlowCollector<T>.() -> T) = flow {
+    runCatching { block() }
+        .onSuccess { emit(it) }
+        .onFailure { throw it}
+}
```

Comment:
Not sure why you decided to remove the whole `mapError()` section. However from the template level, we should have that to form the way to handle custom exceptions and errors 🙏 

```suggestion
@OptIn(ExperimentalTypeInference::class)
fun <T> flowTransform(@BuilderInference block: suspend FlowCollector<T>.() -> T) = flow {
    runCatching { block() }
        .onSuccess { emit(it) }
        .onFailure { throw it.mapError() }
}

private fun Throwable.mapError(): Throwable =
    when (this) {
    	// TODO catch custom errors
        else -> this
    }
```